### ISSUE TITLE
ceph-salt issues wrong "ceph orch" command and gets "Error: 2 hosts provided, expected 3"

### ISSUE THREAD

**Issue by smithfarm** [ceph/ceph-salt#101]
Very odd. . .

sesdev command line:

```
sesdev create octopus --ceph-salt-repo https://github.com/smithfarm/ceph-salt.git --ceph-salt-branch wip-fix-broken-osd-deploy --ceph-container-image="registry.opensuse.org/filesystems/ceph/master/upstream/images/ceph/ceph" --no-deploy-mons --no-deploy-mgrs --no-deploy-osds octopus_test1
```

Results in:

```
    admin:   |   o- Mgr .......................................................................................................... [Minions: 3]
    admin:   |   | o- node1.octopus_test1.com .............................................................................. [other roles: mon]
    admin:   |   | o- node2.octopus_test1.com .............................................................................. [other roles: mon]
    admin:   |   | o- node3.octopus_test1.com .............................................................................. [other roles: mon]
...
    admin:   o- Deployment .............................................................................................................. [...]
    admin:   | o- Bootstrap ......................................................................................................... [enabled]
    admin:   | o- Dashboard ............................................................................................................. [...]
    admin:   | | o- password ............................................................................................................ [***]
    admin:   | | o- username .......................................................................................................... [admin]
    admin:   | o- Mgr ............................................................................................................... [enabled]
    admin:   | o- Mon .............................................................................................................. [disabled]
    admin:   | o- OSD .............................................................................................................. [disabled]
```

and:

```
    admin: [2020-02-25 16:49:11.286782] [node1.octopus_te] Finished with failures
    admin: Failure in minion: node1.octopus_test1.com
    admin: __id__: deploy remaining mgrs
    admin: __run_num__: 55
    admin: __sls__: ceph-salt.ceph-mgr
    admin: changes:
    admin:   pid: 14979
    admin:   retcode: 22
    admin:   stderr: "Error EINVAL: Traceback (most recent call last):\n  File \"/usr/share/ceph/mgr/mgr_module.py\"\
    admin:     , line 1070, in _handle_command\n    return self.handle_command(inbuf, cmd)\n\
    admin:     \  File \"/usr/share/ceph/mgr/orchestrator/_interface.py\", line 191, in handle_command\n\
    admin:     \    return dispatch[cmd['prefix']].call(self, cmd, inbuf)\n  File \"/usr/share/ceph/mgr/mgr_module.py\"\
    admin:     , line 309, in call\n    return self.func(mgr, **kwargs)\n  File \"/usr/share/ceph/mgr/orchestrator/_interf
ace.py\"\
    admin:     , line 153, in <lambda>\n    wrapper_copy = lambda *l_args, **l_kwargs: wrapper(*l_args,\
    admin:     \ **l_kwargs)\n  File \"/usr/share/ceph/mgr/orchestrator/_interface.py\", line\
    admin:     \ 144, in wrapper\n    return func(*args, **kwargs)\n  File \"/usr/share/ceph/mgr/orchestrator/module.py\"\
    admin:     , line 668, in _apply_mgr\n    completion = self.apply_mgr(spec)\n  File \"/usr/share/ceph/mgr/orchestrator
/_interface.py\"\
    admin:     , line 1694, in inner\n    completion = self._oremote(method_name, args, kwargs)\n\
    admin:     \  File \"/usr/share/ceph/mgr/orchestrator/_interface.py\", line 1764, in _oremote\n\
    admin:     \    return mgr.remote(o, meth, *args, **kwargs)\n  File \"/usr/share/ceph/mgr/mgr_module.py\"\
    admin:     , line 1432, in remote\n    args, kwargs)\nRuntimeError: Remote method threw exception:\
    admin:     \ Traceback (most recent call last):\n  File \"/usr/share/ceph/mgr/cephadm/module.py\"\
    admin:     , line 2155, in apply_mgr\n    len(spec.placement.hosts), num_new_mgrs))\nRuntimeError:\
    admin:     \ Error: 2 hosts provided, expected 3"
    admin:   stdout: ''
    admin: comment: 'Command "ceph orch apply mgr 3 node2 node3
    admin:
    admin:   " run'
    admin: duration: 403.004
    admin: name: 'ceph orch apply mgr 3 node2 node3
    admin:
    admin:   '
    admin: result: false
    admin: start_time: '17:49:10.863246'
    admin: state: 'cmd_|-deploy remaining mgrs_|-ceph orch apply mgr 3 node2 node3
    admin:
    admin:   _|-run'
    admin: Finished execution of ceph-salt formula
    admin: Summary: Total=4 Succeeded=3 Failed=1
Command '['vagrant', 'up']' failed: ret=1 stderr:
```

**Comment by smithfarm** [comment]
Implemented a fix in #99 - with that in place, I get:

```
    admin: [2020-02-25 17:40:15.101849] [node1.octopus_te] [STAGE] [BEGIN] Deployment of Ceph MGRs
    admin: [2020-02-25 17:40:21.508056] [node1.octopus_te] [STAGE] [END  ] Deployment of Ceph MGRs
    admin: [2020-02-25 17:40:21.532100] [node1.octopus_te] Finished successfully
    admin: Finished execution of ceph-salt formula
    admin: Summary: Total=4 Succeeded=4 Failed=0
=== Deployment Finished ===

You can login into the cluster with:

  $ sesdev ssh octopus_test1

Or, access the Ceph Dashboard with:

  $ sesdev tunnel octopus_test1 dashboard
```

**Comment by smithfarm** [comment]
Now I'm getting the same error for `ceph orch apply mon`, so I added a commit implementing the same fix in `ceph-mon.sls`
